### PR TITLE
Make returning subscriptions optional

### PR DESCRIPTION
**Pull request by pklinef**
Enables the use of catalog methods in cases where the email address might not be available.

### CODE REVIEW

File: src/test/java/com/connexta/ddf/persistence/subscriptions/SubscriptionMethodsTest.java [hunk 1]
```diff
@@ -40,9 +40,10 @@ public void setUp() throws Exception {
     }
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void testEmptyEmail() throws Exception {
-    testSubscription.getSubscriptions("");
+  @Test
+  public void testEmptyEmail() {
+    List<String> results = testSubscription.getSubscriptions("");
```
Comment (mojogitoverhere) [inline src/test/java/com/connexta/ddf/persistence/subscriptions/SubscriptionMethodsTest.java]: ✏️ Should we also add a test for the null case while we're at it?
